Examples/Holocron-iOSExample/ViewController.swift

```diff
@@ -7,127 +7,122 @@
 
 import UIKit
 import Holocron
-import Result
+
+extension Key {
+    static let fileTest: Key = "fileTestKey"
+    static let keychainTest: Key = "testKeychainKey"
+    static let userDefaultsTest: Key = "testUserDefaultsKey"
+}
 
 class ViewController: UIViewController {
-    
-    lazy var userDefaultsPersistence: UserDefaultsPersistence = UserDefaultsPersistence()
-    let userDefaultsStore = UserDefaultsStore(key: "testUserDefaults")
-    
-    lazy var keychainPersistence: KeychainPersistence = KeychainPersistence(keychainServiceName: "com.holocron.test")
-    let keyChainStore = KeychainStore(key: "testKeychainPersistence")
-    
-    let cacheDirectory = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first!
-    lazy var filePersistance: FilePersistence = FilePersistence(directory: cacheDirectory)
-    let fileStore = FileStore(fileName: "testFilePersistance")
-    
+    let defaultsProvider: StorageProvider = UserDefaults.standard
+    let keychainProvider: StorageProvider = .Keychain(service: "com.holocron.example")
+    let fileProvider = FileSystemStorageProvider(baseURL: FileManager.default.temporaryDirectory)
+
     @IBOutlet private var userDefaultsTextField: UITextField!
     @IBOutlet private var keychainTextField: UITextField!
     @IBOutlet private var fileTextField: UITextField!
-    
+
+    struct Stored: Codable {
+        let title: String
+    }
 
     // MARK: User Defaults
     @IBAction func userDefaultsReadTapped(_ sender: Any) {
```

Comment:
Kinda nitpick, but can you change the `Any` to be `UIButton` for all the @IBActions that are buttons?

Reply:
I'll just get rid of the `sender` param.